jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/ValidationExtension.java

```diff
@@ -53,6 +54,16 @@ public String getName()
         return "@validation";
     }
 
+    @Override
+    public void setConfiguration(Configuration configuration)
+    {
+        super.setConfiguration(configuration);
+
+        if (!(configuration instanceof WebSocketCoreSession))
+            throw new IllegalArgumentException("ValidationExtension needs a CoreSession Configuration");
+        coreSession = (WebSocketCoreSession)configuration;
+    }
```

Comment:
Yuck! If this class (which is internal) needs `WebSocketCoreSession`, give that, not a configuration.
The whole concept that `WebSocketCoreSession` _is-a_ configuration is IMHO wrong.

Reply:
The wrongness here is that if extensions need to know the coreSession, they should be able to get it without needing to down cast a configuration.
Why isn't this method just `setCoreSession` rather than `setConfiguration`?

Reply:
I could change it to `setCoreSession` using the API `CoreSession`, but for `ValidationExtension` we would still need to cast that to the implementation class `WebSocketCoreSession` if we want to get access to these validation methods which aren't in the `CoreSession` interface.

Reply:
So is that saying we actually need a `Validator` class with static methods that take both a Configuration and a Frame, which can then be used in both places without down cast?